source/app.d

```diff
@@ -26,41 +26,52 @@ DubRegistry s_registry;
 DubRegistryWebFrontend s_web;
 string s_mirror;
 
+void checkForNewVersions()
+{
+	if (s_mirror.length) s_registry.mirrorRegistry(URL(s_mirror));
+	else s_registry.updatePackages();
+}
+
 void startMonitoring()
 {
 	void monitorPackages()
 	{
 		sleep(10.seconds()); // give the cache a chance to warm up first
 		while(true){
-			if (s_mirror.length) s_registry.mirrorRegistry(URL(s_mirror));
-			else s_registry.updatePackages();
+			checkForNewVersions;
 			sleep(30.minutes());
 		}
 	}
 	s_checkTask = runTask(&monitorPackages);
 }
 
 version (linux) private immutable string certPath;
-
 shared static this()
 {
+	enum debianCA = "/etc/ssl/certs/ca-certificates.crt";
+	enum redhatCA = "/etc/pki/tls/certs/ca-bundle.crt";
+	immutable certPath = redhatCA.exists ? redhatCA : debianCA;
+}
+
+void main()
+{
 	setLogFile("log.txt", LogLevel.diagnostic);
 
 	string hostname = "code.dlang.org";
+	bool separateCron;
+	bool runCron;
 
 	readOption("mirror", &s_mirror, "URL of a package registry that this instance should mirror (WARNING: will overwrite local database!)");
 	readOption("hostname", &hostname, "Domain name of this instance (default: code.dlang.org)");
+	readOption("separate-cron", &separateCron, "Use a separate cron job to query for packages.");
```

Comment:
We could also remove this flag as the dyno must likely gets suspended and one or two additional updates won't be problematic.